src/index.tsx

```diff
@@ -9,126 +9,132 @@ function getValue(target: any, { isCheckbox }: { isCheckbox: boolean }) {
   return target ? (isCheckbox ? target.checked : target.value) : target;
 }
 
-const RHFInput = React.memo(
-  ({
-    setValue: setValueFromProp,
-    register: registerFromProp,
-    unregister: unregisterFromProp,
-    name,
-    rules,
-    mode = 'onSubmit',
-    as,
-    onChange,
-    onBlur,
-    type,
-    value,
-    defaultValue,
-    defaultChecked,
-    onChangeName,
-    onChangeEvent,
-    onBlurName,
-    onBlurEvent,
-    ...rest
-  }: Props) => {
-    const isCheckbox = type === 'checkbox';
-    const isOnChange = mode === 'onChange';
-    const isOnBlur = mode === 'onBlur';
-    const defaultData = isCheckbox
-      ? defaultChecked === undefined
-        ? false
-        : defaultChecked
-      : defaultValue === undefined
-      ? ''
-      : defaultValue;
-    const [inputValue, setInputValue] = React.useState(defaultData);
-    const valueRef = React.useRef(defaultData);
-    const methods = useFormContext() || {};
-    const setValue = setValueFromProp || methods.setValue;
-    const register = registerFromProp || methods.register;
-    const unregister = unregisterFromProp || methods.unregister;
+const RHFInput = <T,>({
```

Comment:
```suggestion
const RHFInput = <T>({
```

Reply:
This will fail. The reason for comma in `<T,>` is to tell TS that it is a type and not a JSX.

Reply:
Why are we removing `React.memo`?

Reply:
> This will fail. The reason for comma in `<T,>` is to tell TS that it is a type and not a JSX.

is that the case? oh i didn't know that.

Reply:
> Why are we removing `React.memo`?

did we? we shouldn't. I will spend my weekend to test your PR, i will release a beta version for this as well. really good stuff @JeromeDeLeon

Reply:
I see. You're the one who wrote the `React.memo` so I thought you have reason for it.

Reply:
If you think we should remove it then I'm good :smile: